src/ui_fsmenu/addons/manager.cc

```diff
@@ -849,9 +913,48 @@ bool AddOnsCtrl::handle_key(bool down, SDL_Keysym code) {
 	return UI::Window::handle_key(down, code);
 }
 
+void AddOnsCtrl::erase_remote(std::shared_ptr<AddOns::AddOnInfo> a) {
+	for (auto& pointer : remotes_) {
+		if (pointer == a) {
+			pointer = remotes_.back();
+			remotes_.pop_back();
+			return;
+		}
+	}
+	NEVER_HERE();
+}
+
 void AddOnsCtrl::refresh_remotes() {
+	UI::ProgressWindow progress(this, "", "");
+	const std::string step_message = _("Fetching add-ons (%.1f%%)");
+
 	try {
-		remotes_ = net().refresh_remotes();
+		progress.step(_("Connecting to the server…"));
+
+		std::vector<std::string> names = net().refresh_remotes(net().is_admin());
+
+		const int64_t nr_orig_entries = names.size();
+		int64_t nr_addons = nr_orig_entries;
+		remotes_.resize(nr_addons);
+
+		for (int64_t i = 0, counter = 0; i < nr_addons; ++i, ++counter) {
+			progress.step((boost::format(step_message) % (100.0 * counter / nr_orig_entries)).str());
+
+			try {
+				remotes_[i].reset(new AddOns::AddOnInfo(net().fetch_one_remote(names[i])));
+
+				if (!remotes_[i]->matches_widelands_version()) {
+					throw WLWarning("", "incompatible Widelands version");
+				}
```

Comment:
This makes the `all` argument in `refresh_remotes` useless. Incompatible add-ons are not filtered by the server but here by the client.